diff --git a/node_modules/@shopify/hydrogen/dist/development/index.js b/node_modules/@shopify/hydrogen/dist/development/index.js
index 4ea2dc1..2058ec3 100644
--- a/node_modules/@shopify/hydrogen/dist/development/index.js
+++ b/node_modules/@shopify/hydrogen/dist/development/index.js
@@ -212,6 +212,7 @@ function useCustomerPrivacy(props) {
   return result;
 }
 var hasEmitted = false;
+
 function emitCustomerPrivacyApiLoaded() {
   if (hasEmitted) return;
   hasEmitted = true;
