diff --git a/node_modules/@shopify/hydrogen/dist/development/index.js b/node_modules/@shopify/hydrogen/dist/development/index.js
index 4ea2dc1..2db6df8 100644
--- a/node_modules/@shopify/hydrogen/dist/development/index.js
+++ b/node_modules/@shopify/hydrogen/dist/development/index.js
@@ -29,7 +29,11 @@ function AnalyticsView(props) {
     shop
   };
   useEffect(() => {
-    if (!shop?.shopId) return;
+    console.log("[h2:debug:AnalyticsView] - Publishing view event", type, viewPayload2);
+    if (!shop?.shopId) {
+      console.error('[h2:error:AnalyticsView] - shopId is required for analytics events. Make sure to pass the shop prop to <Analytics.Provider> or use <Analytics.Provider shop={shop}>.');
+      return;
+    }
     viewPayload2 = {
       ...viewPayload2,
       url: window.location.href
@@ -80,21 +84,35 @@ function logMissingConfig(fieldName) {
   );
 }
 function useCustomerPrivacy(props) {
+  const { subscribe: subscribe2, register: register2, canTrack } = useAnalytics();
+  const { ready: customerPrivacyReady } = register2("Customer_Privacy");
   const {
     withPrivacyBanner = false,
     onVisitorConsentCollected,
     onReady,
     ...consentConfig
   } = props;
-  useLoadScript(withPrivacyBanner ? CONSENT_API_WITH_BANNER : CONSENT_API, {
+  const status = useLoadScript(withPrivacyBanner ? CONSENT_API_WITH_BANNER : CONSENT_API, {
     attributes: {
       id: "customer-privacy-api"
     }
   });
+  console.log('[debug:useCustomerPrivacy] - useLoadScript status:', status);
   const { observing, setLoaded } = useApisLoaded({
     withPrivacyBanner,
-    onLoaded: onReady
+    onLoaded: (event) => {
+      customerPrivacyReady()
+      onReady(event)
+    }
   });
+  function emitCustomerPrivacyApiLoaded() {
+    if (hasEmitted) return;
+    hasEmitted = true;
+    const event = new CustomEvent("shopifyCustomerPrivacyApiLoaded", {bubbles: true, cancelable: true});
+    document.dispatchEvent(event);
+    console.log('[debug:useCustomerPrivacy] - Emitting shopifyCustomerPrivacyApiLoaded event');
+  }
+
   const config = useMemo(() => {
     const { checkoutDomain, storefrontAccessToken } = consentConfig;
     if (!checkoutDomain) logMissingConfig("checkoutDomain");
@@ -157,9 +175,11 @@ function useCustomerPrivacy(props) {
     withPrivacyBanner,
     config,
     overridePrivacyBannerMethods,
-    setLoaded.privacyBanner
+    setLoaded.privacyBanner,
+    status
   ]);
   useEffect(() => {
+    console.log('[debug:useCustomerPrivacy] - useEffect: Observing customerPrivacy', window?.Shopify);
     if (observing.current.customerPrivacy) return;
     observing.current.customerPrivacy = true;
     let customCustomerPrivacy = null;
@@ -183,26 +203,28 @@ function useCustomerPrivacy(props) {
                 customCustomerPrivacy = {
                   ...customerPrivacy,
                   setTrackingConsent: overrideCustomerPrivacySetTrackingConsent(
-                    { customerPrivacy, config }
-                  )
-                };
-                customShopify = {
-                  ...customShopify,
-                  customerPrivacy: customCustomerPrivacy
-                };
-                setLoaded.customerPrivacy();
-                emitCustomerPrivacyApiLoaded();
-              }
+                  { customerPrivacy, config }
+                )
+              };
+              customShopify = {
+                ...customShopify,
+                customerPrivacy: customCustomerPrivacy
+              };
+              setLoaded.customerPrivacy();
+              emitCustomerPrivacyApiLoaded();
             }
-          });
-        }
+          }
+        });
       }
-    });
-  }, [
-    config,
-    overrideCustomerPrivacySetTrackingConsent,
-    setLoaded.customerPrivacy
-  ]);
+    }
+  });
+}, [
+  config,
+  overrideCustomerPrivacySetTrackingConsent,
+  setLoaded.customerPrivacy,
+  emitCustomerPrivacyApiLoaded,
+  status
+]);
   const result = {
     customerPrivacy: getCustomerPrivacy()
   };
@@ -212,12 +234,6 @@ function useCustomerPrivacy(props) {
   return result;
 }
 var hasEmitted = false;
-function emitCustomerPrivacyApiLoaded() {
-  if (hasEmitted) return;
-  hasEmitted = true;
-  const event = new CustomEvent("shopifyCustomerPrivacyApiLoaded");
-  document.dispatchEvent(event);
-}
 function useApisLoaded({
   withPrivacyBanner,
   onLoaded
@@ -230,8 +246,10 @@ function useApisLoaded({
   const setLoaded = {
     customerPrivacy: () => {
       if (withPrivacyBanner) {
+        console.log('Customer Privacy API loaded');
         setApisLoaded((prev) => [true, prev[1]]);
       } else {
+        console.log('Customer Privacy API loaded without Privacy Banner');
         setApisLoaded(() => [true]);
       }
     },
@@ -239,6 +257,7 @@ function useApisLoaded({
       if (!withPrivacyBanner) {
         return;
       }
+      console.log('Privacy Banner API loaded');
       setApisLoaded((prev) => [prev[0], true]);
     }
   };
@@ -346,8 +365,14 @@ function ShopifyAnalytics({
     locale: language,
     checkoutDomain: !checkoutDomain ? "mock.shop" : checkoutDomain,
     storefrontAccessToken: !storefrontAccessToken ? "abcdefghijklmnopqrstuvwxyz123456" : storefrontAccessToken,
-    onVisitorConsentCollected: () => setPrivacyReady(true),
-    onReady: () => setPrivacyReady(true)
+    onVisitorConsentCollected: () => {
+      setPrivacyReady(true)
+      console.log('useCustomerPrivacy: Customer Privacy API loaded, onVisitorConsentCollected called');
+    },
+    onReady: () => {
+      console.log('useCustomerPrivacy: Customer Privacy API loaded, onReady called');
+      setPrivacyReady(true)
+    }
   });
   useShopifyCookies({
     hasUserConsent: privacyReady ? canTrack() : true,
@@ -357,18 +382,23 @@ function ShopifyAnalytics({
   });
   useEffect(() => {
     if (init.current) return;
+
     init.current = true;
     subscribe2(AnalyticsEvent.PAGE_VIEWED, pageViewHandler);
     subscribe2(AnalyticsEvent.PRODUCT_VIEWED, productViewHandler);
     subscribe2(AnalyticsEvent.COLLECTION_VIEWED, collectionViewHandler);
     subscribe2(AnalyticsEvent.SEARCH_VIEWED, searchViewHandler);
     subscribe2(AnalyticsEvent.PRODUCT_ADD_TO_CART, productAddedToCartHandler);
+    console.log('ShopifyAnalytics: Subscribed to analytics events, initialized');
     setShopifyReady(true);
   }, [subscribe2]);
   useEffect(() => {
     if (shopifyReady && privacyReady) {
       shopifyAnalyticsReady();
       onReady();
+      console.log(
+        'ShopifyAnalytics: Shopify Analytics API loaded'
+      )
     }
   }, [shopifyReady, privacyReady, onReady]);
   return null;
@@ -716,6 +746,7 @@ function PerfKit({ shop }) {
     if (scriptStatus !== "done" || loadedEvent.current) return;
     loadedEvent.current = true;
     subscribe2(AnalyticsEvent.PAGE_VIEWED, () => {
+      console.log("[h2:debug:PerfKit] - Page viewed event received, navigating PerfKit");
       window.PerfKit?.navigate();
     });
     subscribe2(AnalyticsEvent.PRODUCT_VIEWED, () => {
@@ -756,6 +787,7 @@ var defaultAnalyticsContext = {
   customData: {},
   prevCart: null,
   publish: () => {
+    console.log('defaultAnalyticsContext: publish is no-op because canTrack is no-op by default');
   },
   shop: null,
   subscribe: () => {
@@ -776,12 +808,16 @@ function areRegistersReady() {
 function subscribe(event, callback) {
   if (!subscribers.has(event)) {
     subscribers.set(event, /* @__PURE__ */ new Map());
+    console.log(`subscribe: created new event ${event} in subscribers map`);
   }
+  console.log(`subscribe: adding callback for event ${event}`, callback.toString());
   subscribers.get(event)?.set(callback.toString(), callback);
 }
 var waitForReadyQueue = /* @__PURE__ */ new Map();
 function publish(event, payload) {
+  console.log('actual publish: areRegistersReady()?', areRegistersReady());
   if (!areRegistersReady()) {
+    console.log(`publish: ${event} is waiting for registers to be ready`);
     waitForReadyQueue.set(event, payload);
     return;
   }
@@ -790,6 +826,7 @@ function publish(event, payload) {
 function publishEvent(event, payload) {
   (subscribers.get(event) ?? /* @__PURE__ */ new Map()).forEach((callback, subscriber) => {
     try {
+      console.log(`publishEvent: ${event} to ${subscriber}`);
       callback(payload);
     } catch (error) {
       if (typeof error === "object" && error instanceof Error) {
@@ -811,6 +848,7 @@ function register(key) {
   }
   return {
     ready: () => {
+      console.log(`registered: ${key} is ready`, waitForReadyQueue.size, 'events in queue');
       registers[key] = true;
       if (areRegistersReady() && waitForReadyQueue.size > 0) {
         waitForReadyQueue.forEach((queuePayload, queueEvent) => {
@@ -847,8 +885,16 @@ function AnalyticsProvider({
   );
   const [carts, setCarts] = useState({ cart: null, prevCart: null });
   const [canTrack, setCanTrack] = useState(
-    customCanTrack ? () => customCanTrack : () => shopifyCanTrack
+    customCanTrack ? () => customCanTrack : () => function shopifyCanTrack() {
+      try {
+        return window.Shopify.customerPrivacy.analyticsProcessingAllowed();
+      } catch (e) {
+        console.error('[h2:error:Analytics.Provider:shopifyCanTrack] - Shopify Customer Privacy API not yet available.');
+      }
+      return false;
+    }
   );
+
   if (!!shop) {
     if (/\/68817551382$/.test(shop.shopId)) {
       warnOnce(
@@ -880,34 +926,71 @@ function AnalyticsProvider({
       }
     }
   }
+
+  function getCustomerPrivacy() {
+    try {
+      return window.Shopify && window.Shopify.customerPrivacy ? window.Shopify?.customerPrivacy : null;
+    } catch (e) {
+      return null;
+    }
+  }
+
+  function getPrivacyBanner() {
+    try {
+      return window && window?.privacyBanner ? window.privacyBanner : null;
+    } catch (e) {
+      return null;
+    }
+  }
+
+  const p = canTrack()
+    ? publish
+    : () => {
+        console.log('AnalyticsProvider: publish is no-op because canTrack is false')
+      }
+
   const value = useMemo(() => {
     return {
       canTrack,
       ...carts,
       customData,
-      publish: canTrack() ? publish : () => {
-      },
+      publish: p,
       shop,
       subscribe,
       register,
       customerPrivacy: getCustomerPrivacy(),
       privacyBanner: getPrivacyBanner()
-    };
-  }, [
-    analyticsLoaded,
-    canTrack,
-    carts,
-    carts.cart?.updatedAt,
-    carts.prevCart,
-    publish,
-    subscribe,
-    customData,
-    shop,
-    register,
-    JSON.stringify(registers),
-    getCustomerPrivacy,
-    getPrivacyBanner
-  ]);
+    },
+    [
+        canTrack,
+        carts,
+        customData,
+        p,
+        shop,
+        getCustomerPrivacy,
+        getPrivacyBanner,
+        register,
+        subscribe,
+        getCustomerPrivacy,
+        getPrivacyBanner
+    ];
+  })
+
+  console.log('-------------------------')
+  console.log(JSON.stringify(value))
+  console.log('-------------------------')
+
+  // listen to shopifyCustomerPrivacyApiLoaded custom event
+  useEffect(() => {
+    function shopifyCustomerPrivacyApiLoaded(event) {
+      console.log('--> privacy api loaded event', event);
+    }
+    window.addEventListener('shopifyCustomerPrivacyApiLoaded', shopifyCustomerPrivacyApiLoaded);
+
+    return () => {
+      window.removeEventListener('shopifyCustomerPrivacyApiLoaded', shopifyCustomerPrivacyApiLoaded);
+    }
+  }, []);
   return /* @__PURE__ */ jsxs(AnalyticsContext.Provider, { value, children: [
     children,
     !!shop && /* @__PURE__ */ jsx(AnalyticsPageView, {}),
@@ -918,9 +1001,16 @@ function AnalyticsProvider({
         consent,
         onReady: () => {
           listenerSet.current = true;
+          console.log('ShopifyAnalytics:onReady shopifyCanTrack', typeof shopifyCanTrack, shopifyCanTrack());
           setAnalyticsLoaded(true);
           setCanTrack(
-            customCanTrack ? () => customCanTrack : () => shopifyCanTrack
+            customCanTrack ? () => customCanTrack : () => function canTrack() {
+              try {
+                return window.Shopify.customerPrivacy.analyticsProcessingAllowed();
+              } catch (e) {
+              }
+              return false;
+            }
           );
         },
         domain: cookieDomain
@@ -929,6 +1019,7 @@ function AnalyticsProvider({
     !!shop && /* @__PURE__ */ jsx(PerfKit, { shop })
   ] });
 }
+
 function useAnalytics() {
   const analyticsContext = useContext(AnalyticsContext);
   if (!analyticsContext) {
